api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -571,7 +576,15 @@ public static void expandJar(File fileToExpand, File tmpModuleDir, String name,
 					String entryName = jarEntry.getName();
 					// trim out the name path from the name of the new file
 					if (!keepFullPath && name != null) {
-						entryName = entryName.replaceFirst(name, "");
+						if (name.equals(entryName)) {
+							int lastSlash = name.lastIndexOf('/');
```

Comment:
Yes, I ran unit tests also on Windows and they do not fail, which proves that paths in `JarEntry` do not use File.Separator but always forward slash.